src/Sema.zig

```diff
@@ -3218,6 +3213,12 @@ fn zirValidateArrayInit(
             break :inst block.instructions.items[block_index + 1];
         };
 
+        // Array has one possible value, so value is always comptime-known
```

Comment:
This usage of OPV seems like it might be incorrect for tuples, since they can have OPV without that being shared by all their elements.

I kept it all the same for now, since this change is just to make sure that the calculation of `first_block_index` is not skipped.

I think we can tackle the array/tuple init differences in another PR, along with #11182